src/loader/bgw_launcher.c

```diff
@@ -842,6 +845,35 @@ database_is_template_check(void)
 	ReleaseSysCache(tuple);
 }
 
+/*
+ * Before we morph into the scheduler, we also need to reload configs from
```

Comment:
I would make the interaction with waiting on the commit more clear here. This function really exists to reload the settings after the txn being waited on has completed because we rely on changes to the timescaledb.restoring flag.